taichi/backends/cuda/cuda_device.h

```diff
@@ -80,19 +81,19 @@ class CudaDevice : public Device {
     void *ptr{nullptr};
     size_t size{0};
     bool is_imported{false};
+    bool use_cached{false};
   };
 
   AllocInfo get_alloc_info(DeviceAllocation handle);
 
   ~CudaDevice() override{};
 
   DeviceAllocation allocate_memory(const AllocParams &params) override;
-  DeviceAllocation allocate_memory_runtime(const AllocParams &params,
-                                           JITModule *runtime_jit_module,
-                                           LLVMRuntime *runtime,
-                                           uint64 *result_buffer) override;
+  DeviceAllocation allocate_memory_runtime(
+      const LlvmRuntimeAllocParams &params) override;
```

Comment:
Ideally packing things into struct `LlvmRuntimeAllocParams` should be a separate PR which can be approved pretty quicky ;)